### PR TITLE
Synthesis and P&R: Move closer to OpenROAD-flow-scripts

### PR DESCRIPTION
**Pull request by mtdudek**
Synthesis:
This commit adds a custom abc.script from OpenROAD-flow-scripts. This commit also modifies the synthesis script. It adds 2 new steps: extract adders - it tries to map addition chains to PDK specific full adder and half adder improving overall performance. Runs only when `adder_mapping` attr is provided.
tie logic cells - it runs `hilomap` pass.

P&R:
This commit changes output_db file extention to .odb. OpenROAD file explorer only allows for .odb to be opened from GUI.

Allow user to disable `improve_placement` pass during resize step. It can cause cell overlapping both by not maintaining sufficient cell to cell spacing as well as true cell overlapping.
Some other minor changes to the P&R scripts.

Edit:
Related to #239

### CODE REVIEW

**Review by mithro — COMMENTED** [review]
This change is going to be a bit complicated to land as it affects the results on Google internal benchmarks. It would be good if we could land this in multiple parts.

**Review by mithro — COMMENTED** [review]
Sorry about this, but I think we need this pull request split into the following;
 1) `enable_improve_placement` functionality (should be quick to merge)
 2) `.db` into `.odb` (should be quick to merge)
 3) `estimate_parasitics` addition (should be quick to merge)
 4) `set_propagated_clock` & `detailed_route` addition (may cause issues)
 5) `remove_buffers` change  (may cause issues)
 6) Tie cell & hi/low change  (may cause issues)
 7) Adder mapping  (may cause issues)
 8) ABC script.  (may cause issues)

Would you be able to do that?

File: place_and_route/open_road.bzl [hunk 1]
```diff
@@ -210,9 +210,9 @@ def openroad_command(ctx, commands, input_db = None, step_name = None, inputs =
```
[the remark this answers is not in the document]
Comment (mtdudek) [reply place_and_route/open_road.bzl:210]: No problem, I'll split it into smaller changes

**Comment by mithro** [comment]
I'm going to close this pull request as we are landing it in parts.

**Pull request closed without merging**